Feature/integration

srslie opened the pull request:
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [x] Fixes a bug
- [x] Cleans up (logs, tests, etc.)


## *Description: What does this PR do?*
- Integrates charts and forms into app
- Adds minor styling


## *What are the relevant tickets?*
Closes #12,  Closes #1,  Closes #3, 


## *Bugs:*
- still some conditional/async compiler errors while waiting for data in the component
- need to style especially canvas sizes
- need to navigate to new pages after form submission/delete/etc with reload

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have run Lighthouse/Wave checks for accessibility
- [ ] New and existing unit tests pass locally with my change

camaragon committed:
Install chart.js and react chart.js

camaragon committed:
Setup Circle Chart component

camaragon committed:
Import the circle chart component

alia-peterson committed:
Start new course form component

camaragon committed:
Add setup for goals form

camaragon committed:
Add a recommended object to mock data

camaragon committed:
Display chart with the mock data

camaragon committed:
Install sass

camaragon committed:
Install cypress

camaragon committed:
Setup add goals form

camaragon committed:
Setup goals form sass file

camaragon committed:
Add CircleChart cypress file

camaragon committed:
Add goalsForm cypress file

camaragon committed:
Finish setup for goals form

camaragon committed:
Add styling to goals form

srslie committed:
Add activites data

srslie committed:
Add instruction spec and tests

srslie committed:
Add module form spec outline

srslie committed:
Add basic instructions page

srslie committed:
Hardcode inputs

srslie committed:
Make boolean lowercase

srslie committed:
Add names and metric

alia-peterson committed:
Add hooks and submit functionality to form component

alia-peterson committed:
Create button mixin

alia-peterson committed:
Add form styling

alia-peterson committed:
Update form styling

srslie committed:
Add helper functions and styling

srslie committed:
Add basic styling

camaragon committed:
Delete pseudocode for circle chart

srslie committed:
Add sass

camaragon committed:
Add sass file for circle chart

camaragon committed:
Add tag for total hours

alia-peterson committed:
Update newCourse object per BE

camaragon committed:
Setup dynamic rendering for circle chart

camaragon committed:
Filter data by mod

srslie committed:
Migrate styling to scss

srslie committed:
Add syling, remove course choice

srslie committed:
Add activities get

srslie committed:
Add activities from api

camaragon committed:
Add logic to the circle chart

srslie committed:
Add bar functionality

srslie committed:
Add trigger of context fetch

alia-peterson committed:
Add goal lookup functionality

alia-peterson committed:
Update input fields to select dropdowns

srslie committed:
Add unique module name validation

srslie committed:
Fix current module bug

srslie committed:
Add homepage link to logo

alia-peterson committed:
Add post functionality for new course

alia-peterson committed:
Create horizonal chart component

alia-peterson committed:
Update course dashboard to display horizontal charts

alia-peterson committed:
Add functionality to find percent of time per activity

camaragon committed:
Install chartjs dependencies

camaragon committed:
Update calculations methods for charts
Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>

camaragon committed:
Update module dashboard with useState and useEffect
Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>

camaragon committed:
Delete the circleChart tag

camaragon committed:
Refactor course dashboard and useEffect
Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>

camaragon committed:
Delete some mock data

camaragon committed:
Update app context functionality
Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>

alia-peterson committed:
Update charts to have tooltips on hover

camaragon committed:
Add delete data api fetch
Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>

camaragon committed:
Refactor circleChart component

camaragon committed:
Change h2 text

srslie committed:
Add time and styling

camaragon committed:
Make barChart and circleChart visible
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>

camaragon committed:
Comment out calculations test for CI

camaragon committed:
Pass test for CI

srslie committed:
Play with nav styling

srslie committed:
Change background color of form

srslie committed:
Merge branch 'main' into feature/integration

srslie committed:
Update dependencies

srslie committed:
Merge remote-tracking branch 'origin/cma/circle-chart' into feature/integration

srslie committed:
Merge remote-tracking branch 'origin/feature/module-activities-chart' into feature/integration

srslie committed:
Add basic styling file, arrangment

srslie committed:
Add colors to activities

srslie committed:
Denote inprogress tests

Co-authored-by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon <69489633+camaragon@users.noreply.github.com>

srslie committed:
Delete outdated files

Co-authored-by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon <69489633+camaragon@users.noreply.github.com>

alia-peterson commented on components/CircleChart.js:
```diff
@@ -1,10 +1,55 @@
+        data: [...chartData],
+        backgroundColor: [
+          "#FF6384",
+          "#36A2EB",
```
We can put these into a different file so that all of us can access the same colors across different charts

alia-peterson commented on components/HorizontalChart.js:
```diff
@@ -0,0 +1,63 @@
+
+export default function HorizontalChart({ activities }) {
+  // const { sharedState, setSharedState } = useAppContext()
+  // const { courseActivities, setCourseActivities } = useState({})
```
I think these can be removed later, still deciding how the data is being stored/handled for this graph

alia-peterson commented on components/NavBar.js:
```diff
@@ -46,11 +58,11 @@ export default function NavBar() {
         <div className={styles.courseButtons}>
-          <Link href='/moduleDashboard'>
+          {/* <Link href='/moduleDashboard'>
             <div className= {styles.module}>
```
Looks like this can be deleted now since we decided allmodules is redundant with the course

alia-peterson commented on components/NavBar.js:
```diff
@@ -67,30 +79,31 @@ export default function NavBar() {
       <section className={styles.courseButtons}>
         <Link href='/'>
-          <div className={styles.course}>
+          <div className={styles.course, styles.home}>
```
I don't believe a comma separation for these classnames actually works, I think it only takes the first style. this should work though: `className={'${styles.course} ${styles.home}'}`
(the quotes inside the first set of brackets will need to be backticks but the formatting was weird for the comment so i made them single quotes)

alia-peterson commented on components/NavBar.js:
```diff
@@ -67,30 +79,31 @@ export default function NavBar() {
 
       <section className={styles.navButtons}>
+        <Link href='/addCourseForm'>
+          <div className={styles.addCourse, styles.courseButtons}>
```
same here for formatting the classnames

alia-peterson commented on context/app-context.js:
```diff
@@ -1,46 +1,58 @@
-                })
-            })
-    }, []);
+        console.log('TRIGGERD', Date.now(), hasBeenUpdated)
```
can remove the console.log here

srslie committed:
Clean up calculations

Co-authored-by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon <69489633+camaragon@users.noreply.github.com>

alia-peterson commented on pages/addCourseForm.js:
```diff
@@ -1,5 +1,130 @@
+
 export default function addCourseForm() {
+  const { sharedState, setSharedState } = useAppContext()
+  // const { hasBeenUpdated, setHasBeenUpdated } = useAppContext()
```
I'll add this in now that it's available with this merge

camaragon commented on components/BarChart.js:
```diff
@@ -1,7 +1,7 @@
 import { Bar } from 'react-chartjs-2';
 import { useAppContext } from '../context/app-context';
-import { formatDataForBarChart } from '../utilities/calculations';
+import { calculations } from '../utilities/calculations';
```
I reformatted the calculations file, so I changed the imports

camaragon commented on context/app-context.js:
```diff
@@ -1,46 +1,58 @@
+        sharedState,
         setSharedState,
-        sharedState
+        hasBeenDeleted,
```
There is more data accessible in context now

alia-peterson committed:
Remove unused classnames and commented link

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

camaragon reviewed (approved):
This all looks great and it's nice to have all the components connected. Let's finish testing and fix these bugs

alia-peterson reviewed (approved):
- [x] Changes generate no new warnings or errors
- [x] Branch was tested locally without failures (sort of, since we still have things that we know we need to fix)
- [ ] All tests are passing as expected

alia-peterson committed:
Add key to child element in map

srslie committed:
Fix setState bug on course recall for new modules

Co-authored-by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon <69489633+camaragon@users.noreply.github.com>

srslie committed:
Merge branch 'feature/integration' of github.com:CourseChart/course-chart-fe into feature/integration

lbmerchant93 commented:
Looks great!
